howto/finetune_lora.md

```diff
@@ -0,0 +1,88 @@
+# Finetuning with LoRA
+
+[Low-rank adaption (LoRA)](https://arxiv.org/abs/2106.09685) is a technique to approximate the update to the linear layers in a LLM with a low-rank matrix factorization. This significantly reduces the number of trainable parameters and speeds up training with little impact on the final performance of the model.
+We demonstrate this method by instruction-finetuning Lit-Parrot StableLM 3B on the [Alpaca](https://github.com/tatsu-lab/stanford_alpaca) dataset on a **single RTX 3090 (24GB) GPU**.
+
+## Preparation
+
+The steps here only need to be done once:
+
+1. Follow the instructions in the [README](../README.md) to install the dependencies.
+2. Download and convert the weights and save them in the `./checkpoints` folder as described [here](download_weights.md).
+3. Download the data and generate the instruction tuning dataset:
+
+   ```bash
+   python scripts/prepare_alpaca.py
+   ```
+
+See also: [Finetuning on an unstructured dataset](unstructured_dataset.md)
```

Comment:
This file doesn't exist in this repo

Reply:
Good catch, I linked Aniket's blog post for now.